index.src.html

```diff
@@ -1810,73 +1818,269 @@
+        {{TypeError}}.
+    4. Let |resource| be the [=media resource=] referenced by
+        {{AudioData/[[resource reference]]}}.
+    5. Let |channelBytes| be the region of |resource| corresponding to
+        |channelNumber|.
+    6. Copy the |channelBytes| into |destination|.
 
-### Attributes ###{#audioframe-attributes}
-<dl>
-  <dt><dfn attribute for=AudioFrame>timestamp</dfn></dt>
-  <dd>The presentation timestamp, given in microseconds.</dd>
+: <dfn method for=AudioData>clone()</dfn>
+:: Creates a new AudioData with a reference to the same [=media resource=].
 
-  <dt><dfn attribute for=AudioFrame>buffer</dfn></dt>
-  <dd>The buffer containing decoded audio data.</dd>
-</dl>
+    When invoked, run these steps:
+    1. If the value of |frame|'s {{AudioData/[[detached]]}} internal slot is
+        `true`, throw an {{InvalidStateError}} {{DOMException}}.
+    2. Return the result of running the [=Clone AudioData=] algorithm with
+        [=this=].
 
+: <dfn method for=AudioData>close()</dfn>
+:: Clears all state and releases the reference to the [=media resource=].
+    Close is final.
 
-### Methods ###{#audioframe-methods}
-<dl>
-  <dt><dfn method for=AudioFrame>close()</dfn></dt>
-  <dd>
-    Immediately frees [=system resources=]. When invoked, run these steps:
-    1. Release [=system resources=] for buffer and set its value to null.
-    2. Assign `true` to the {{AudioFrame/[[detached]]}} internal slot.
+    When invoked, run these steps:
+    1. Assign `true` to the {{AudioData/[[detached]]}} internal slot.
+    2. Assign `null` to {{AudioData/[[resource reference]]}}.
+
+### Algorithms ### {#audiodata-algorithms}
+
+: <dfn>Clone AudioData</dfn> (with |data|)
+:: Run these steps:
+    1. Let |clone| be a new {{AudioData}} initialized as follows:
+        1. Let |resource| be the [=media resource=] refrenced by |data|'s
+            {{AudioData/[[resource reference]]}}.
+        2. Let |reference| be a new reference to |resource|.
+        3. Assign |reference| to {{AudioData/[[resource reference]]}}.
+        4. Assign the values of |data|'s {{AudioData/[[detached]]}},
+            {{AudioData/[[sample format]]}}, {{AudioData/[[sample rate]]}},
+            {{AudioData/[[number of frames]]}},
+            {{AudioData/[[number of channels]]}}, and
+            {{AudioData/[[timestamp]]}} slots to the corresponding slots in
+            |clone|.
+    2. Return |clone|.
+
+Audio Sample Format{#audio-sample-format}
+-----------------------------------------
+Audo sample formats describe the numeric type used to represent a single
+sample (e.g. 32-bit floating point) and the arrangement of samples from
+different channels as either interleaved or planar.
 
-    NOTE: This section needs work. We should use the name and semantics of
-        VideoFrame destroy(). Similarly, we should add clone() to make a deep
-        copy.
-  </dd>
-</dl>
+<xmp class='idl'>
+enum AudioSampleFormat {
+  "U8",
```

Comment:
Do we want to continue RIFF WAVE's tradition of using u8 but signed for all other integer widths? What about endianess?

Reply:
This is probably fine for now. I was thinking that maybe we could expose single-byte samples as `i8` because it's more comfortable to do DSP type stuff when the bias is at 0, but it's quite established really.